cpp/throwaways/generate_gnd_association_and_spelling_replacements.sh

```diff
@@ -0,0 +1,93 @@
+# Create krimdok local keyword file from the original output and make misspellings and potential GND references explicit
+#!/bin/bash
+
+
+if [ $# != 4 ]; then
+    echo "usage: $0 A-Z_output A_Z_bereinigt gnd_full.mrc result"
+    exit 1
+fi
+
+readonly AZ_OUTPUT=$1
+readonly AZ_CLEANED=$2
+readonly GND_FULL=$3
+readonly RESULT_FILE=$4
+readonly tmp_prefix=$(basename $0)
+#readonly gnd_matches="./matched3.txt"
+#readonly gnd_matches="./tmp/matched_210609_1.txt"
+
+function MkTempFile {
+  echo $(mktemp /tmp/${tmp_prefix}.XXXXX)
+}
+
+
+function GenerateKeywordToGNDMap {
+   readonly gnd_matched=$(MkTempFile)
+   readonly gnd_unmatched=$(MkTempFile)
+   readonly gnd_matched_clean=$(MkTempFile)
+   single_marc_convert_keywords ${GND_FULL} ${AZ_CLEANED} ${gnd_matched} ${gnd_unmatched} 
+   cat ${gnd_matched} | sed -r 's/^"|"$//g' |  sed 's/","/;/' > ${gnd_matched_clean}
+}
+
+
+function ReorderAZOutput {
+     readonly az_output_reordered=$(MkTempFile)
+     # Remove ';' at the end and normalize \r\n
+     cat ${AZ_OUTPUT} | sed ':a;N;$!ba;s/;\r\n/\n/g' | sed '$ s/.$//' | \
+     `# Resort columns to get final form in first column` \
+     tr -d '\r' |  awk -F';' '{ 
+          printf("%s", $NF); 
+          for (i = 1; i < NF; i++) { 
+              printf(";%s", $i); 
+          } 
+          print ""; 
+     }' > ${az_output_reordered} 
+}
+
+
+function JoinAssociationAndSpelling {
+    readonly association_and_spelling=$(MkTempFile)
```

Comment:
This should be define outside the function, because it is also used in at least 1 other function later